src/odb/src/defin/definReader.cpp

```diff
@@ -72,6 +72,107 @@
   }
 namespace odb {
 
+namespace {
+
+// Helper function to get the correct number of bits of a cell for scandef.
+int calculateBitsForCellInScandef(const int bits, dbInst* inst)
+{
+  // -1 is no bits were provided in the scandef
+  if (bits != -1) {
+    return bits;
+  }
+  // -1 means that the bits are not set in the scandef
+  // We need to check if the inst is sequential to decide what is a
+  // reasonable default value
+  dbMaster* master = inst->getMaster();
+  if (master->isSequential()) {
+    // the default number of bits for sequential elements is 1
+    return 1;
+  }
+  // the default number of bits for combinational logic is 0
+  return 0;
+}
+
+dbITerm* findScanITerm(definReader* reader,
+                       dbInst* inst,
+                       const char* pin_name,
+                       const char* common_pin)
+{
+  if (!pin_name) {
+    if (!common_pin) {
+      reader->_logger->error(utl::ODB,
+                             259,
+                             "SCANDEF is missing either component pin or a "
+                             "COMMONSCANPINS for instance {}",
+                             inst->getName());
```

Comment:
Errors should be handled though definReader::error (same throughout).  Note that it will not stop execution and you should be prepared to continue (generates more than the first error, also -continue_on_errors).

Reply:
Thanks for the suggestion!, I think I got your comment about continuing showing more than the first error, I tried to do that when possible. Thanks!, this is ready for another look.